lib/p2p/NodeList.ts

```diff
@@ -141,6 +160,9 @@ class NodeList extends EventEmitter {
       } else if (node.banned) {
         // If the reputationScore is not below the banThreshold but node.banned
         // is true that means that the node was unbanned
+        if (this.inBannedList(node.nodePubKey)) {
+          this.removeBannedList(node.nodePubKey);
```

Comment:
Here for example, I think we can remove the node from the list within the `setBanStatus` method, and we shouldn't need to check twice if the node is banned.